usaspending_api/awards/v2/filters/sub_award.py

```diff
@@ -224,14 +226,17 @@ def recip_string_parse(recipient_string):
             queryset = queryset.filter(q) if q else queryset
 
         # add "naics_codes" (column naics) after NAICS are mapped to subawards
-        elif key in ("program_numbers", "contract_pricing_type_codes"):
-            filter_to_col = {
-                "program_numbers": "cfda_number",
-                "contract_pricing_type_codes": "type_of_contract_pricing",
-            }
-            in_query = [v for v in value]
-            if len(in_query) != 0:
-                queryset &= SubawardView.objects.filter(**{"{}__in".format(filter_to_col[key]): in_query})
+        elif key in ("contract_pricing_type_codes"):
+            if len(value) != 0:
+                queryset &= SubawardView.objects.filter(type_of_contract_pricing__in=value)
+
+        elif key == "program_numbers":
+            if len(value) != 0:
+                queryset &= SubawardView.objects.filter(
+                    award__in=Subquery(
+                        TransactionFABS.objects.filter(cfda_number__in=value).values("award_id")
```

Comment:
TransactionFABS has no concept of `award_id` so this won't work. Will need to either use TransactionSearch or a combination of TransactionNormalized and TransactionFABS.